releases/stable-branches-and-release-cycle.md

```diff
@@ -0,0 +1,113 @@
+for stable branches or bugfix releases. In the following proposal we define a plan
+to make this possible starting with 0.8.
+
+## Proposal
+
+### Alignment of branch -> image releases
+
+We produce currently `<image>:devel` for every master push, along with `<image>:x.y.z` for
+the releases.
+
+Changing the `master` branch name to `devel` would make this homogeneous so our image releases
+are simply `<image>:<branch-or-tag>`.
+
+### Creating one stable branch for every minor release
+
+Approaching the first release candidate `rc0` of every minor or major version a new stable
+branch will be forked off `devel` branch in the form of `release-x.y` by the release scripts,
+and a patch or PR pinning the shipyard-base-image to `release-x.y` will be applied by the
+automated process.
+
+Any following changes related to pinning go modules in submariner, admiral, lighthouse, etc...
+will be done in the `release-x.y` branch directly (no cherry picks).
+
+Once the branch exists, any patches which should be part of the release will need to be
+merged in `devel`and backported to `release-x.y`.
+
+### Maintaining a stable branch
+
+#### Latest stable branch (N)
+
+The only patches that should be submitted to the latest stable branch are
+bugfixes and not features, unless for exceptions previously agreed on the
+community meeting.
+
+Any patches that are necessary for fixing bugs in a stable branch will
+need to be merged first into `devel`, and then cherry-picked to the stable
+branch.
+
+`release-0.9` <-- `devel` <-- `bug-fix patch`
+
+*No-change version bumps* will be done if a base image has been updated
+due to a security bugfix, our code would remain the same, but our versions
+would be bumped to pickup the new images.
+
+#### Previous stable branch (N-1)
+
+The only patches that should be submitted to the previous stable branch (N-1)
+are critical bugfixes, being the definition fo critical bugfix: security
+related fixes, and CVEs.
+
+The patches necessary for the critical bugfixes should be cherry-picked from
+previous  versions, though sometimes they could need specific adaptations
+to the branch.
+
+`release-0.8`  <-- `release-0.9` <-- `devel` <-- `bug-fix patch`
+
+*No-change version bumps* will be necessary if a base image has been updated
+due to a security bugfix, our code would remain the same, but our versions
+would be bumped to pickup the new images.
+
+#### Releasing a stable branch
+
+When a patch is merged to a stable branch the images for such branch will be updated, for
+example `quay.io/submariner/submariner:release-x.y`.
+
+This image allows for testing a set of bugfix patches, sometimes multiple patches are expected
+to be required for one fix. Once the release brach is ready we should create a new x.x.**x**
+patch  release via the release repository.
+
+### How long is a stable branch maintained
+
+Initially stable branches (N) will be maintained until the next stable
```

Comment:
Is 'N' a single Stable Branch or a number of stable branches?

Reply:
ohh it's quite unclear, I meant current stable branch = N, previous = N-1 let me fix it.